exercises/forth/cases_test.go

```diff
@@ -0,0 +1,223 @@
+package forth
+
+// Source: exercism/x-common
+// Commit: 8804a76 Remove non-ASCII characters tests from forth exercise
+
+var testCases = []testCase{
+	// parsing and numbers
+	{
+		"empty input results in empty stack",
+		[]string{},
+		[]int{},
+	},
+	{
+		"numbers just get pushed onto the stack",
+		[]string{"1 2 3 4 5"},
+		[]int{1, 2, 3, 4, 5},
+	},
+	{
+		"all non-word characters are separators",
+		[]string{"1\x002\x133\n4\r5 6\t7"},
+		[]int{1, 2, 3, 4, 5, 6, 7},
+	},
+
+	// addition
+	{
+		"can add two numbers",
+		[]string{"1 2 +"},
+		[]int{3},
+	},
+	{
+		"errors if there is nothing on the stack",
```

Comment:
In an earlier version, I had used multiple testCases, one for additionTestCases, subtractionTestCases, and so forth, because the canonical-data.json (read by gen.go and parsed by the template example_gen.go) was "sectioned". I was able to output the section "name", like "addition" when an error occurred so it addressed these duplicate instances of "error if nothing on the stack".  It did make the test program longer since it used different TestXxxx functions, although each called a "runTest" helper to actually do the mechanics. When running go test, the multiple TestXxxx allowed multiple errors to occur, failing at most one in each "section".  I thought it best to keep only one error in front of the student at one time.  So, I redid the text template in example_gen to put all the test cases together, and that made go test stop on first error, allowed a very simple forth_test and also a simple benchmark function as well.  Long story for this, but the multiple "errors if there is nothing on the stack" originates from the descriptions in canonical-data.json.  I'm thinking of two options:  1) Use the "sectioned" test cases again to allow the error output to indicate which section of the failed case, or 2) Update the canonical-data.json in x-common/exercises/forth to use more specific descriptions, like "+ causes error if nothing is on the stack", and "dup causes error if nothing is on the stack".   I think I like option 2 better(different pull request for that repo), but can easily be convinced to do option 1, since I have that code on a different branch already. Also, is there is a way to stop on first error when you have multiple TestXxxx functions?  Maybe that is a minor thing, as I think I encountered it while "exercising" with go.